refactor(skills): replace arch-diagram-builder agent with reusable architecture-diagrams skill (#2068)
microsoft/hve-core: .github/skills/project-planning/adr-author/templates/diagram-mermaid.md

Variant: mermaid
Selected when: state.userPreferences.diagramFormat == "mermaid"
Purpose: lightweight starting scaffold for Mermaid flowchart diagrams in ADRs

> IaC-derived diagrams: when the ADR diagram is derived from infrastructure source files (Terraform, Bicep, ARM), author it with the architecture-diagrams skill (.github/skills/hve-core/architecture-diagrams/SKILL.md) using mermaid output and embed that result in place of this scaffold; that skill owns the Mermaid conventions, arrow types, and subgraph boundaries.

Reviewer: [portability] Hardcoded skill path in template metadata comment

Same issue as `diagram-ascii.md`: the comment references `.github/skills/hve-core/architecture-diagrams/SKILL.md`, which does not resolve in plugin or extension distributions.

```suggestion
IaC-derived diagrams: when the ADR diagram is derived from infrastructure source files (Terraform, Bicep, ARM), use the `architecture-diagrams` skill (load its SKILL.md) with mermaid output and embed that result in place of this scaffold; that skill owns the Mermaid conventions, arrow types, and subgraph boundaries.
```

Author: Done. Same fix as `diagram-ascii.md`: the comment now references the `architecture-diagrams` skill by name rather than the repo-root path, so it resolves in plugin and extension distributions. Pushed in 057e8313.
